doc/man3/RSA_public_encrypt.pod

```diff
@@ -46,23 +48,32 @@ Encrypting user data directly with RSA is insecure.
 
 =back
 
-B<flen> must be less than RSA_size(B<rsa>) - 11 for the PKCS #1 v1.5
-based padding modes, less than RSA_size(B<rsa>) - 41 for
+B<flen> must not be more than RSA_size(B<rsa>) - 11 for the PKCS #1 v1.5
+based padding modes, not more than RSA_size(B<rsa>) - 42 for
 RSA_PKCS1_OAEP_PADDING and exactly RSA_size(B<rsa>) for RSA_NO_PADDING.
-The random number generator must be seeded prior to calling
-RSA_public_encrypt().
+When a padding mode other than RSA_NO_PADDING is in use, then
+RSA_public_encrypt() will include some random bytes into the ciphertext
+and therefore the ciphertext will be different each time, even if the
+plaintext and the public key are exactly identical.
+B<to> and B<from> may overlap.
 
 RSA_private_decrypt() decrypts the B<flen> bytes at B<from> using the
-private key B<rsa> and stores the plaintext in B<to>. B<to> must point
-to a memory section large enough to hold the decrypted data (which is
-smaller than RSA_size(B<rsa>)). B<padding> is the padding mode that
-was used to encrypt the data.
+private key B<rsa> and stores the plaintext in B<to>. B<flen> should
+be equal to RSA_size(B<rsa>) but may be smaller, when leading zero bytes
+are in the ciphertext. However this is not recommended. B<to> must point
```

Comment:
It's unclear what you're saying is not recommended. As far as I know, leading 0's in the ciphertext are not important at all and can be removed. What we want to hide is the length of the plain text, and so when checking the padding of the decrypted text, it should be padded with 0's, but that is nothing the user of this function controls.

Reply:
I guess there is also about a 1/256 chance that there is a leading 0 in the ciphertext.

Reply:
Yes, 1/256, unless you have RSA-2049 then it is 1/2.
Do we have test coverage for this removed padding?

Reply:
So I made my point now clearer.